pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -58,10 +63,17 @@ const (
 	requiredMCPCertCheckFreq = 500 * time.Millisecond
 )
 
+var useBuiltinGalleyConfigSource = env.RegisterBoolVar("USE_BUILTIN_GALLEY_SOURCE", false,
+	"Use the built-in Galley's processing pipeline as the source of configuration ")
+
 // initConfigController creates the config controller in the pilotConfig.
 func (s *Server) initConfigController(args *PilotArgs) error {
 	meshConfig := s.environment.Mesh()
-	if len(meshConfig.ConfigSources) > 0 {
+	if useBuiltinGalleyConfigSource.Get() {
+		if err := s.initInprocessGalleyController(args); err != nil {
+			return err
+		}
+	} else if len(meshConfig.ConfigSources) > 0 {
```

Comment:
isn't this just yet another way of using galley as the source of data in istiod when pilot already supports reading directly from k8s? What is the rationale for this addition, i.e. the feature that Pilot lacks but Galley has in its mcp implementation?

Reply:
And btw, pilot can have multiple config sources (galley plus others). So this if-else does not seem right, in fact it breaks systems taht use multiple sources.

Reply:
Good question. The rationale was unifying the configuration processing between Galley and Pilot and eliminating duplicate code. 

* Duplicated config transformations, e.g. k8s ingress, [native k8s gateway resource](https://kubernetes-sigs.github.io/service-apis/) (soon). For example, we're already sharing the transformation logic between Galley's config distribution and istioctl analyze. 

* Duplicate config source watching logic, e.g. k8s/file/mcp. These should behave the same regardless if the user is running Galley+Pilot or Istiod.

* Lack of CRD status. Galley config pipeline is responsible for updating CRD status today.

* Eventual support for directly peering Istio control planes across environments/clusters via MCP/UDPA. 

@nmittler has already made several changes to unify the schemas between Galley and Pilot. This PR lets us experiment with the e2e integration and identify further areas for perf improvement. This is all flag protected so we don't affect any existing non-experimental configuration.

Reply:
@nmittler suggested writing up an RFC for this line of work. I think that makes sense and is something we can review in the networking WG.

Reply:
re: multiple sources of config, Pilot has a single source for user config and multiple sources for service/endpoints. IMO the code is correct in the sense that we can have only one source for user config. The confusing part is that we've overloaded the MCP config controller for endpoints when that part really belongs in the initServiceRegistryController path. That is something we'll need to fix.

Reply:
For what its worth, since you mentioned it, I gave up writing the k8s gateway stuff because the galley code is so much more complicated to do simple transformations (compare pilot Ingress vs galley Ingress)